stunnel/stunnel.ml

```diff
@@ -344,33 +344,29 @@ let connect
     | None -> () in
 	retry (fun () -> attempt_one_connect ?unique_id ?use_fork_exec_helper ?write_to_log _verify_cert extended_diagnosis host port) 5
 
-let sub_after i s =
-  let len = String.length s in
-    String.sub s i (len - i)
-
-let split_1 c s =
-  match String.split ~limit:1 c s with
-    | x :: _ -> x
-    | [] -> s
-
 let check_verify_error line =
-  match String.find_all "VERIFY ERROR: " line with
-      | p :: _ ->
-          begin
-            match String.find_all "error=" line with
-              | e :: _ ->
-                  raise
-                    (Stunnel_verify_error
-                       (split_1 ','
-                          (sub_after (e + String.length "error=") line)))
-              | [] ->
-                  raise (Stunnel_verify_error "")
-          end
-      | [] ->
-          ()
+  let sub_after i s =
+    let len = String.length s in
+    String.sub s i (len - i)
+  in
+  let split_1 c s =
+    match Astring.String.cut ~sep:c s with
+    | Some (x , _) -> x
+    | None -> s
+  in
+  if Astring.String.is_infix ~affix:"VERIFY ERROR: " line then
+    match Astring.String.find_sub ~sub:"error=" line with
+    | Some e ->
+      raise
+        (Stunnel_verify_error
+           (split_1 ","
+              (sub_after (e + String.length "error=") line)))
```

Comment:
Not sure if this can be written differently with the Astring substring functions, but its fine as it is.

Reply:
It probably can, I didn't think too much about it. i think we should have an action to review and refactor stunnel, I have seen lots of things that could be simplified while I was making the jbuilder port. But again I think it should be done separately and undergo lots of testing